Documentation/grpc-metadata.md

```diff
@@ -223,3 +223,60 @@ func (s *server) SomeStreamingRPC(stream pb.Service_SomeStreamingRPCServer) erro
     stream.SetTrailer(trailer)
 }
 ```
+
+## Updating metadata from a server interceptor
+
+An example for updating metadata from a server interceptor is
+available [here](../examples/features/metadata_interceptor/server/main.go).
+
+#### Unary interceptor
```

Comment:
Can we move these sections to the new README instead, to avoid clutter here and also to put the documentation near the example that illustrates it?

Reply:
@dfawley  Thank you very much for your response. Do you mean moving the `Updating metadata from a server interceptor` part of `grpc-metadata.md` to the `README.md` under the folder `examples/features/metadata_interceptor`? Please correct me if anything I am misunderstanding or missing.

Reply:
I mean move everything from `#### Unary interceptor` down into `examples/features/metadata_interceptor/README.md`.

Reply:
@dfawley , thank you very much for your response. I have updated the document per your review comment. Could you please review it again?